heudiconv/convert.py

```diff
@@ -806,3 +820,23 @@ def  add_taskname_to_infofile(infofiles):
 
         # write to outfile
         save_json(infofile, meta_info)
+
+
+def bvals_are_zero(bval_file, threshold=5):
+    """Checks if all entries in a bvals file are zero (or below the threshold).
+    Returns True if that is the case, otherwise returns False
+
+    Parameters
+    ----------
+    bval_file : file with the bvals
+    threshold : b-value below which they are considered to be zero. (Default: 5)
```

Comment:
posted against some original commit : do you have immediate usecases where they are <5 but not 0 and should still be considered?

Reply:
No, not `<5`, but yes `=5`.
As I mentioned, for Siemens scanners with relatively up-to-date software versions (I have tested it for VE11C), when you select a b-value of 0, it ends up writing `5` in the header (it takes into account slice-encoding gradients and readout gradients).

**Usecase:**
(Tested in Siemens scanners): If you want to collect a pair of Spin-Echo (SE) EPI images (`pepolar`) to estimate the `fmap`, the best way to ensure that all the readout parameters of the DW images you are going to correct are matched is by making a copy of the DW sequence, set the number of b-values to 1, and set this b-value to zero. That way you have the same FoV, matrix size, acceleration factor, partial-Fourier and echo spacing. (Then, I set the TE to the minimum allowed by the sequence, to get higher SNR, but that is not necessary.) Because the b-value is zero, there are no diffusion-weighting gradients, and the resulting image is a SE image.
The resulting images have `'DIFFUSION'` in the `image_type` list, and a b-value of 5, and need to be extracted as `fmap`.
I have found cases that, for some reason or other, if I started with a standard SE sequence and tried to modify the sequence parameters to match those of the diffusion-weighted sequence, I haven't been able to. I don't remember exactly when it happens, but it could have been that I hadn't been able to match the echo-spacing, or the matrix size, or something else. But if you just start with a copy of the DW sequence, you are guaranteed to have same readout.

Reply:
hm, should we then make it like 
```python
bvals_unique = set(float(b) for b in bvals)
return bvals_unique == {0.} or bvals_unique == {5.}
```
to be "exact" and not silently succumb to some other odd cases?

Reply:
Sure.

Reply:
Resolved in 5099d01a086af776f454c92446331428b69d67e6